wso2/testgrid
# Changes needed when using the TestGrid local deployment

**Purpose**

Package all components of Testgrid - TG-Java, Jenkins, Dashboard, Database(H2) and then make it executable on local Linux instances.
![Blank Diagram(1)](https://user-images.githubusercontent.com/19789134/65035812-c2870c00-d967-11e9-9ca4-ff5053ab171a.jpeg)

**Sample User Story**

1.  Get the localTestgrid zip file and unzip it

2.  Move into the unzipped folder and enter the credentials in credentials.config file

3.  Move into the testgrid-home folder and enter configuration properties in configurations.properties file

4.  Open the terminal within the unzipped folder and execute command ./local-tg startup-jenkins

     (Then Tomcat server will be started and all the plugins needed will be installed in Jenkins engine. The very first time it will take about 10 minutes. Then the credentials will be added to Jenkins engine and H2 server will be started )

5.  Access Jenkins engine at localhost:8080/jenkins URL

     (User will see testgrid-job-creator job listed in there)

6.  When user build the testgrid-job-creator job all other jobs currently in the https://testgrid-live-dev.private.wso2.com will be listed in the Jenkins dashboard

7.  Then the user can build any of the jobs listed in the Jenkins dashboard.

8.  Execute command ./local-tg shutdown-jenkins to shutdown Tomcat server and H2 server.
